src/third-party/arduino-json-5.2/include/ArduinoJson/Arduino/String.hpp

```diff
@@ -9,11 +9,8 @@
 
 #ifndef ARDUINO
 
-#ifndef ARDUINO_STRING_OVERRIDE

```

Comment:
The other option that I think would work (been a while since I have been directly building large projects) but I think if we built arduino-json as a library than it wouldn't matter if there was a naming conflict, their code would use their typedef and our code would use our override.

I need to do more research on that.